src/Ocelot/Requester/MessageInvokerPool.cs

```diff
@@ -58,9 +58,11 @@ private HttpMessageInvoker CreateMessageInvoker(DownstreamRoute downstreamRoute)
 
         // Adding timeout handler to the top of the chain.
         // It's standard behavior to throw TimeoutException after the defined timeout (90 seconds by default)
-        var timeoutHandler = new TimeoutDelegatingHandler(downstreamRoute.QosOptions.TimeoutValue == 0
-            ? TimeSpan.FromSeconds(RequestTimeoutSeconds)
-            : TimeSpan.FromMilliseconds(downstreamRoute.QosOptions.TimeoutValue))
+        var timeoutHandler = new TimeoutDelegatingHandler(downstreamRoute.QosOptions.TimeoutValue > 0
```

Comment:
I'm not sure about the logic here. The `ITimeoutCreator` will return you a timeout value, it includes the default timeout value too.

Reply:
I apologize, but which `ITimeoutCreator` are you referring to? We don't have such an interface in our solution.
**Could you provide a code snippet for the author?**
Perhaps it would be more sensible to integrate the business logic into the existing `TimeoutDelegatingHandler` class, wouldn't it? If we modify this logic, wouldn't it necessitate a review of numerous tests?

In my view, the [TimeoutDelegatingHandler](https://github.com/search?q=repo%3AThreeMammals%2FOcelot%20TimeoutDelegatingHandler&type=code) is the ideal place to set the default 90-second timeout value and the new property. We also need to access the global shared value in this class.

My concern is that the `TimeoutDelegatingHandler` isn't a standard service class in DI, and since it's part of the private messaging system logic, it cannot be overridden. However, this is beyond the scope of the current linked issues.